elliottlib/cli/verify_attached_bugs_cli.py

```diff
@@ -126,6 +135,25 @@ def validate(self, non_flaw_bugs: Iterable[Bug], verify_bug_status: bool, no_ver
             exit(1)
         green_print("All bugs were verified. This check doesn't cover CVE flaw bugs.")
 
+    def verify_bugs_advisory_type(self, non_flaw_bugs, advisory_id_map, advisory_bug_map):
+        bugs_by_type = categorize_bugs_by_type(non_flaw_bugs, advisory_id_map)
+        for kind, advisory_id in advisory_id_map.items():
+            if kind in ['metadata', '?']:
+                continue
+            actual = {b for b in advisory_bug_map[advisory_id] if b.is_ocp_bug()}
+            expected = bugs_by_type[kind]
+            if actual != expected:
+                bugs_not_found = expected - actual
+                extra_bugs = actual - expected
+                if bugs_not_found:
+                    self._complain(f'Bugs not found in {kind} advisory ({advisory_id}):'
+                                   f' {[b.id for b in bugs_not_found]}')
+                if extra_bugs:
+                    self._complain(f'Extra bugs in {kind} advisory ({advisory_id}): {[b.id for b in extra_bugs]}')
+        if self.problems:
+            red_print("Some bug problems were listed above. Please investigate.")
+            exit(1)
```

Comment:
Would like to not exit here but continue to run other validations, so we can get all bug problems at once.

Reply:
ack. Refactored this